src/main/java/net/javadiscord/javabot/systems/moderation/ModerationService.java

```diff
@@ -198,21 +201,43 @@ public boolean removeTimeout(Member member, String reason, Member removedBy, Mes
 	 * @param bannedBy The member who is responsible for banning this member.
 	 * @param channel  The channel in which the ban was issued.
 	 * @param quiet    If true, don't send a message in the channel.
-	 * @return Whether the moderator has the permission to ban this member or not.
 	 */
-	public boolean ban(Member member, String reason, Member bannedBy, MessageChannel channel, boolean quiet) {
+	public void ban(Member member, String reason, Member bannedBy, MessageChannel channel, boolean quiet) throws IllegalArgumentException, UserPermissionException {
 		var banEmbed = buildBanEmbed(member, reason, bannedBy);
 		if (canBanUser(member, bannedBy)) {
 			member.getUser().openPrivateChannel().queue(
 					c -> c.sendMessage(config.getBanMessageText()).setEmbeds(banEmbed).queue(),
 					e -> log.info("Could not send Ban Direct Message to User {}", member.getUser().getAsTag())
 			);
-			bannedBy.getGuild().ban(member, BAN_DELETE_DAYS, reason).queue();
+
+			try {
+				bannedBy.getGuild().ban(member, BAN_DELETE_DAYS, reason).queue();
+			} catch (InsufficientPermissionException | HierarchyException e) {
+				throw new UserPermissionException(channel.getJDA().getSelfUser(), e);
+			}
```

Comment:
This does not throw any exceptions. You would need to catch the exception in the queue's lambda